heron/schedulers/src/java/com/twitter/heron/scheduler/marathon/MarathonScheduler.java

```diff
@@ -0,0 +1,222 @@
+      uriObject.put(MarathonConstants.CACHE, false);
+
+      urisNode.add(uriObject);
+    }
+
+    return urisNode;
+  }
+
+  protected ArrayNode getPorts(ObjectMapper mapper) {
+    ArrayNode ports = mapper.createArrayNode();
+
+    for (String portName : MarathonConstants.PORT_NAMES) {
+      ObjectNode port = mapper.createObjectNode();
+      port.put(MarathonConstants.PORT, 0);
+      port.put(MarathonConstants.PROTOCOL, MarathonConstants.TCP);
+      port.put(MarathonConstants.PORT_NAME, portName);
+
+      ports.add(port);
+    }
+
+    return ports;
+  }
+
+  protected String getExecutorCommand(int containerIndex) {
+    List<String> commands = new ArrayList<>();
+    commands.add(Context.executorSandboxBinary(config));
+    commands.add(Integer.toString(containerIndex));
+
+    TopologyAPI.Topology topology = Runtime.topology(runtime);
+    commands.add(topology.getName());
+    commands.add(topology.getId());
+    commands.add(FileUtils.getBaseName(Context.topologyDefinitionFile(config)));
+    commands.add(Runtime.instanceDistribution(runtime));
+    commands.add(Context.stateManagerConnectionString(config));
+    commands.add(Context.stateManagerRootPath(config));
+    commands.add(Context.tmasterSandboxBinary(config));
+    commands.add(Context.stmgrSandboxBinary(config));
+    commands.add(Context.metricsManagerSandboxClassPath(config));
+    commands.add(SchedulerUtils.encodeJavaOpts(TopologyUtils.getInstanceJvmOptions(topology)));
+    commands.add(TopologyUtils.makeClassPath(topology, Context.topologyJarFile(config)));
+    commands.add(MarathonConstants.MASTER_PORT);
+    commands.add(MarathonConstants.TMASTER_CONTROLLER_PORT);
+    commands.add(MarathonConstants.TMASTER_STATS_PORT);
+    commands.add(Context.systemConfigSandboxFile(config));
+    commands.add(Runtime.componentRamMap(runtime));
+    commands.add(SchedulerUtils.encodeJavaOpts(TopologyUtils.getComponentJvmOptions(topology)));
+    commands.add(Context.topologyPackageType(config));
+    commands.add(Context.topologyJarFile(config));
+    commands.add(Context.javaSandboxHome(config));
+    commands.add(MarathonConstants.SHELL_PORT);
+    commands.add(Context.shellSandboxBinary(config));
+    commands.add(MarathonConstants.METRICSMGR_PORT);
+    commands.add(Context.cluster(config));
+    commands.add(Context.role(config));
+    commands.add(Context.environ(config));
+    commands.add(Context.instanceSandboxClassPath(config));
+    commands.add(Context.metricsSinksSandboxFile(config));
+
+    String completeSchedulerProcessClassPath = new StringBuilder()
+        .append(Context.schedulerSandboxClassPath(config)).append(":")
+        .append(Context.packingSandboxClassPath(config)).append(":")
+        .append(Context.stateManagerSandboxClassPath(config))
+        .toString();
+    commands.add(completeSchedulerProcessClassPath);
+    commands.add(MarathonConstants.SCHEDULER_PORT);
+
+    return Joiner.on(" ").join(commands);
+  }
```

Comment:
This method duplicate most of the code from `SchedulerUtils.executorCommandArgs`. Can we reuse the original method? It may need some refactoring though.